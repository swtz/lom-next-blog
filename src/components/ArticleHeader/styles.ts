import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      font-size: ${theme.font.sizes.xxlarge};
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.large};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
