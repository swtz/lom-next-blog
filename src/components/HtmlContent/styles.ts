import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.xsmall} + 0.2rem);
    line-height: 1.5;

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    code {
      font-family: monospace;
      color: ${theme.colors.secondary};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.medium} 0;
      font-size: ${theme.font.sizes.xsmall};
      width: 100%;
      overflow-x: auto;
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      background: ${theme.colors.mediumGray};
      font-size: ${theme.font.sizes.xsmall};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium} ${theme.spacings.small};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    .media iframe {
      width: 100%;
      height: auto;
      aspect-ratio: 16 / 9;
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      pre {
        font-size: 1.2rem;
      }

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: ${theme.spacings.xlarge} 0;
      }
    }
  `};
`;
