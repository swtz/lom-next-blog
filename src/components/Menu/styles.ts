import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css``}
`;

export const Nav = styled.nav`
  ${() => css``}
`;

export const Logo = styled.div`
  ${() => css``}
`;

export const OpenClose = styled.a`
  ${() => css``}
`;
