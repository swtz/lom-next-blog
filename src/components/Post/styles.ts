import styled, { css } from 'styled-components';

export const Wrapper = styled.article`
  ${() => css``}
`;

export const Container = styled.div`
  ${() => css``}
`;
