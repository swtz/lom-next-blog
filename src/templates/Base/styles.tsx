import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css``}
`;
export const HeaderContainer = styled.div`
  ${() => css``}
`;
export const ContentContainer = styled.div`
  ${() => css``}
`;
export const FooterContainer = styled.div`
  ${() => css``}
`;
