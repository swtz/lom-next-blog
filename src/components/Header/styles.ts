import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    /* padding: ${theme.spacings.xhuge}; */
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.xsmall};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.xsmall} - 1rem);
    }

    @media ${theme.media.lteXSmall} {
      flex-flow: column wrap;

      > div {
        ${HeadingStyles} {
          margin: ${theme.spacings.xsmall} 0
            calc(${theme.spacings.small} - 1rem);
        }
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
