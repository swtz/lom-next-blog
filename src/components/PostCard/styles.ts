import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  text-align: center;

  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: ${theme.spacings.small};
      margin-bottom: ${theme.spacings.small};
      transition: all 300ms ease-in-out;
    }

    a {
      text-decoration: none;
    }

    &:hover ${HeadingStyles} {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
    }
  `}
`;

export const Cover = styled.img`
  ${() => css`
    max-width: 100%;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css``}
`;
