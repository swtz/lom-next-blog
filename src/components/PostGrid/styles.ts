import styled, { css } from 'styled-components';
import { Wrapper as PostCardStyles } from '../PostCard/styles';

export const Wrapper = styled.div`
  ${() => css``}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    justify-content: center;
    align-items: center;
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
    gap: 3rem;
    padding: 1rem;

    ${PostCardStyles} {
      padding: 1rem;
      border: 0.5rem solid ${theme.colors.secondary};
      border-radius: 5%;
      height: 100%;
      transition: border-color 300ms ease-in-out;
    }

    ${PostCardStyles}:hover {
      border-color: ${theme.colors.darkText};
    }
  `}
`;
