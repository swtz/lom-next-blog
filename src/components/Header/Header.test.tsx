import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Header } from '.';

describe('<Header />', () => {
  it('should render', () => {
    renderTheme(
      <Header
        blogName="Ol치, mundo!"
        blogDescription="A short description"
        logo="a.svg"
        showText={false}
      />,
    );
    expect(screen.queryByRole('paragraph')).toHaveStyle('visibility: 0');
  });

  it('should matches to snapshot', () => {
    const { container } = renderTheme(
      <Header
        blogName="Ol치, mundo!"
        blogDescription="A short description"
        logo="a.svg"
      />,
    );

    expect(container.firstChild).toMatchInlineSnapshot(`
      .c3 {
        color: #000000;
        font-size: 3.2rem;
        text-transform: none;
      }

      .c0 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-pack: center;
        -webkit-justify-content: center;
        -ms-flex-pack: center;
        justify-content: center;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        gap: 3.2rem;
      }

      .c2 {
        visibility: 0;
        opacity: 0;
        visibility: 1;
        opacity: 1;
      }

      .c1 {
        width: 100%;
        max-width: 12rem;
      }

      <header
        class="c0"
      >
        <img
          alt="Ol치, mundo!"
          class="c1"
          src="a.svg"
        />
        <div
          class="c2"
        >
          <h3
            class="c3"
          >
            Ol치, mundo!
          </h3>
          <p
            class=""
          >
            A short description
          </p>
        </div>
      </header>
    `);
  });
});
