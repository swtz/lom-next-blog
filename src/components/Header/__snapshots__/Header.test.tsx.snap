// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c2 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 3.2rem 0;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c0 .c1 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 2.4rem;
  margin-right: 2.4rem;
  max-width: 48rem;
}

@media (max-width:550px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    text-align: center;
  }

  .c0 > div .c1 {
    margin: 1.6rem 0 calc(.8rem - 1rem);
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      <a
        class="c3"
        href="/"
        target="_self"
      >
        <img
          alt="Blog do Arroz é vida! - Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa."
          src="/assets/images/logo.svg"
        />
      </a>
    </h1>
    <div
      class="c4"
    >
      <h2
        class="c1 c5"
      >
        Blog do Arroz é vida!
      </h2>
      <p>
        Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa.
      </p>
    </div>
  </header>
</div>
`;
